tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/client/AutoCreateTest.java

```diff
@@ -0,0 +1,312 @@
+         Topic topic = session.createTopic(ADDRESS_NAME);
+         session.createConsumer(topic);
+      }
+
+      { // just a namespace area
+         final AddressInfo infoRef = info;
+         Wait.assertTrue(() -> infoRef.getBindingRemovedTimestamp() != -1);
+      }
+
+      Assert.assertFalse(AssertionLoggerHandler.findText("AMQ224113"));
+      Thread.sleep(50);
+      PostOfficeTestAccessor.reapAddresses((PostOfficeImpl) server.getPostOffice());
+      Assert.assertFalse(AssertionLoggerHandler.findText("AMQ224113"));
+      Assert.assertTrue(info.isSwept());
+      PostOfficeTestAccessor.reapAddresses((PostOfficeImpl) server.getPostOffice());
+      Assert.assertTrue(AssertionLoggerHandler.findText("AMQ224113"));
+   }
+
+
+   @Test
+   public void testNegativeSweepAddress() throws Exception {
+      AssertionLoggerHandler.startCapture();
+      server.getConfiguration().setAddressQueueScanPeriod(-1); // disabling scanner, we will perform it manually
+      AddressSettings settings = new AddressSettings().setAutoDeleteQueues(true).setAutoDeleteAddresses(true).setAutoDeleteAddressesDelay(10).setAutoDeleteQueuesDelay(10);
+      server.getConfiguration().getAddressesSettings().clear();
+      server.getConfiguration().getAddressesSettings().put("#", settings);
+      server.start();
+      String ADDRESS_NAME = "autocreatedmulticast";
+
+      AddressInfo info = new AddressInfo(ADDRESS_NAME).addRoutingType(RoutingType.MULTICAST).setAutoCreated(true);
+      server.getPostOffice().addAddressInfo(info);
+      info = server.getPostOffice().getAddressInfo(SimpleString.toSimpleString(ADDRESS_NAME));
+
+      ConnectionFactory cf = CFUtil.createConnectionFactory("core", "tcp://localhost:61616");
+      try (Connection connection = cf.createConnection()) {
+         Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Topic topic = session.createTopic(ADDRESS_NAME);
+         session.createConsumer(topic);
+      }
+
+      { // just a namespace area
+         final AddressInfo infoRef = info;
+         Wait.assertTrue(() -> infoRef.getBindingRemovedTimestamp() != -1);
+      }
+
+      Assert.assertFalse(AssertionLoggerHandler.findText("AMQ224113"));
+      Thread.sleep(50);
+      PostOfficeTestAccessor.reapAddresses((PostOfficeImpl) server.getPostOffice());
+      Assert.assertFalse(AssertionLoggerHandler.findText("AMQ224113"));
+      Assert.assertTrue(info.isSwept());
+      try (Connection connection = cf.createConnection()) {
+         Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Topic topic = session.createTopic(ADDRESS_NAME);
+         session.createConsumer(topic);
+         PostOfficeTestAccessor.reapAddresses((PostOfficeImpl) server.getPostOffice());
+         Assert.assertFalse(AssertionLoggerHandler.findText("AMQ224113"));
+         Assert.assertFalse(info.isSwept()); // it should be cleared because there is a consumer now
```

Comment:
it should because I disabled the reaper and I am doing it by hand only.

Reply:
You first did the reap before adding the consumer, which means it will be set 'swept' then. Then the next reap is done after the consumer is added and still present, so it will then be set 'not swept' then. I'm questioning if adding the consumer should itself make it not-swept immediately, i.e before the reaper run. It seemed like it would still be marked 'swept' even though the consumer is present, and that it would only be marked 'not swept' if the second reap occurs while the consumer is still present, meaning if the consumer goes away first it will be treated largely the same as if the consumer never existed (delete-delay influence seeming the only difference)

E.g consider a broker with the default 30sec reap config, meaning two unused queues are seen and marked 'swept' at the first reap. These would be be marked 'swept' initially at the same time, and so seems like they could later removed be at the same time even if one remained unused for those next 30s, while the other actually got used [any number of times] much more recently by a new consumer(s) that had been created but then also gone away again inbetween the 2 reaps.

As I say, I can see arguments both ways on this. If it is unused at the time of the reap, and has still passed any delete-delay period (if it has one), then removing it seems fair enough as it satifies being unused at the time. However, it still seems a little inconsistent (as one remained unused, while one was used between the sweeps, meaning its being removed much earlier in its unused state than the other queue) and given that, I wonder what the point is of the reaper ever needing to see it 'unused' twice before actually removing it? (I'm guessing it was so as to not quickly remove unused queues with a 0 value for delete-delay, which still looks to be the default).

Reply:
I see... addQueue will set setSwept (false) now